compiler/rustc_expand/src/expand.rs

```diff
@@ -2021,7 +2022,17 @@ impl<'a, 'b> InvocationCollector<'a, 'b> {
                 }
                 None => {
                     match Node::wrap_flat_map_node_noop_flat_map(node, self, |mut node, this| {
-                        assign_id!(this, node.node_id_mut(), || node.noop_flat_map(this))
+                        assign_id!(this, node.node_id_mut(), || {
+                            let symbols = collect_doc_alias_symbol_from_attrs(node.attrs().iter());
+                            if !symbols.is_empty() {
+                                this.cx.resolver.append_confusable_attr_symbols(
+                                    this.cx.current_expansion.lint_node_id,
+                                    symbols,
+                                );
+                            }
+
+                            node.noop_flat_map(this)
+                        })
```

Comment:
This is not a good place to collect this auxiliary data.

Could you move this to `fn build_reduced_graph_for_item` and similar places?
(Maybe to `fn define` if possible, since you are working with bindings in `lookup_doc_alias_name`.)